src/Phinx/Migration/Manager.php

```diff
@@ -84,35 +84,38 @@ public function printStatus($environment, $format = null)
 
             $env = $this->getEnvironment($environment);
             $versions = $env->getVersions();
-	    try {
-		foreach ($this->getMigrations() as $migration) {
-		    if (in_array($migration->getVersion(), $versions)) {
-			$status = '     <info>up</info> ';
-			unset($versions[array_search($migration->getVersion(), $versions)]);
-		    } else {
-			$status = '   <error>down</error> ';
-		    }
-
-		    $output->writeln(
-			$status
-			. sprintf(' %14.0f ', $migration->getVersion())
-			. ' <comment>' . $migration->getName() . '</comment>'
-		    );
-		    $migrations[] = array('migration_status' => trim(strip_tags($status)), 'migration_id' => sprintf('%14.0f', $migration->getVersion()), 'migration_name' => $migration->getName());
-		}
+            try {
+                foreach ($this->getMigrations() as $migration) {
+                    $isDown = false;
+                    if (in_array($migration->getVersion(), $versions)) {
+                        $status = '     <info>up</info> ';
+                        unset($versions[array_search($migration->getVersion(), $versions)]);
+                    } else {
+                        $isDown = true;
+                        $status = '   <error>down</error> ';
+                    }
+                    if ($showOnlyDownMigrations === false || ($showOnlyDownMigrations === true && $isDown === true)) {
+                        $output->writeln(
+                        $status
+                        . sprintf(' %14.0f ', $migration->getVersion())
+                        . ' <comment>' . $migration->getName() . '</comment>'
+                        );
+                        $migrations[] = array('migration_status' => trim(strip_tags($status)), 'migration_id' => sprintf('%14.0f', $migration->getVersion()), 'migration_name' => $migration->getName());
+                    }
+                }
 
-		foreach ($versions as $missing) {
-		    $output->writeln(
-			'     <error>up</error> '
-			. sprintf(' %14.0f ', $missing)
-			. ' <error>** MISSING **</error>'
-		    );
-		}
-	    } finally {
-	        if ($env->getAdapter() !== null) {
-		        $env->getAdapter()->disconnect();
-	        }
-	    }
+                foreach ($versions as $missing) {
```

Comment:
if a migration is marked as complete in the database, but the code doesn't know about the migration, should it show when the showOnlyDownMigrations is on? Probably a moot point for production.